x/gravity/keeper/msg_server.go

```diff
@@ -323,3 +325,19 @@ func (k msgServer) SubmitBadSignatureEvidence(c context.Context, msg *types.MsgS
 		},
 	)
 }
+
+func (k msgServer) UpdateParams(goCtx context.Context, msg *types.MsgUpdateParams) (*types.MsgUpdateParamsResponse, error) {
+
+	if k.authority != msg.Authority {
+		return nil, errors.Wrapf(govtypes.ErrInvalidSigner, "invalid authority; expected %s, got %s", k.authority, msg.Authority)
+	}
+
+	if err := msg.Params.ValidateBasic(); err != nil {
+		return nil, err
+	}
+
+	ctx := sdk.UnwrapSDKContext(goCtx)
+	k.SetParams(ctx, msg.Params)
```

Comment:
Update `SetParams` to use kvStore

Reply:
I have updated the SetParams. Now it uses the kvStore